public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -120,4 +165,31 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TraceToLo
   getQueryDisplayText(query: TempoQuery) {
     return query.query;
   }
+
+  buildSearchQuery(query: TempoQuery) {
+    const tags = query.search ? query.search.split(/\s+(?=([^"]*"[^"]*")*[^"]*$)/g) : [];
```

Comment:
I wonder if it wouldn't make sense to use the prism syntax to also get the labels and values here. Right now the syntax is only with the `=` operator bu we could define `key` and `value` as tokens for prism (should also give syntax highlight and could be similar to loki/prometheus) and I think there is a prism method to then just get the tokens. Not sure though if that would be particularly easier but seems like we could kill two birds at once with that.

Reply:
I think that'd be a nice improvement although I'll probably need some help. I'll give it a shot and ping the team tomorrow for help depending on where I'm at.

Reply:
Made some improvements to this but there are still some edge cases that aren't handled perfectly that we can iterate on more during the beta testing.